DICOM/src/Microsoft.Health.DeID.SharedLib/RedactFunction.cs

```diff
@@ -0,0 +1,147 @@
+                return DateTimeUtility.IndicateAgeOverThreshold(date)
+                    ? null
+                    : new DateTimeOffset(date.Year, 1, 1, 0, 0, 0, new TimeSpan(0, 0, 0)).ToString(outputDateFormat ?? DeIDGlobalSettings.DateFormat);
+            }
+
+            return null;
+        }
+
+        public string RedactDateTime(string dateTimeString, string inputDateTimeFormat = null, string outputDateFormat = null, IFormatProvider provider = null)
+        {
+            if (string.IsNullOrEmpty(dateTimeString))
+            {
+                return null;
+            }
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                DateTimeOffset date = DateTimeUtility.ParseDateTimeString(dateTimeString, inputDateTimeFormat, provider);
+                return DateTimeUtility.IndicateAgeOverThreshold(date)
+                    ? null
+                    : new DateTimeOffset(date.Year, 1, 1, 0, 0, 0, new TimeSpan(0, 0, 0)).ToString(outputDateFormat ?? DeIDGlobalSettings.DateTimeFormat);
+            }
+
+            return null;
+        }
+
+        public DateTimeOffset? RedactDateTime(DateTimeOffset dateTime)
+        {
+            EnsureArg.IsNotNull<DateTimeOffset>(dateTime, nameof(dateTime));
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                if (DateTimeUtility.IndicateAgeOverThreshold(dateTime))
+                {
+                    return null;
+                }
+
+                return new DateTimeOffset(dateTime.Year, 1, 1, 0, 0, 0, default);
+            }
+
+            return null;
+        }
+
+        public DateTimeObject RedactDateTime(DateTimeObject dateObject)
+        {
+            EnsureArg.IsNotNull(dateObject, nameof(dateObject));
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                if (DateTimeUtility.IndicateAgeOverThreshold(dateObject.DateValue))
+                {
+                    return null;
+                }
+
+                dateObject.DateValue = new DateTimeOffset(dateObject.DateValue.Year, 1, 1, 0, 0, 0, dateObject.HasTimeZone == true ? dateObject.DateValue.Offset : new TimeSpan(0, 0, 0));
+                return dateObject;
+            }
+
+            return null;
+        }
+
+        public int? RedactAge(int age)
+        {
+            return (int?)RedactAge((decimal)age);
+        }
+
+        public decimal? RedactAge(decimal age)
+        {
+            if (_redactSetting.EnablePartialAgesForRedact)
+            {
+                if (age > AgeThreshold)
+                {
+                    return null;
+                }
+
+                return age;
+            }
+
+            return null;
+        }
+
+        public AgeObject RedactAge(AgeObject age)
+        {
+            EnsureArg.IsNotNull(age, nameof(age));
+
+            return RedactAge(age.AgeInYears()) == null ? null : age;
```

Comment:
🐞 [Bug]

In most case it returns a unchanged `age` object.

Reply:
OK, based on offline discussion, it's not a bug but a feature 😅 
In some conditions the age or other element don't need redaction.